prependShellScript option results in non-portable script-JARs

I'm using version 0.14.3.

Because the `prependShellScript` option uses `IO.writelines`, the line terminators used in the shell script in the assembled JAR are platform-dependent. This makes the resulting script non-portable (the JAR itself is OK).

To reproduce, prepend the `defaultShellScript` **on Windows**, then execute the script/JAR on a Unix-y system.

1. Add the following to `build.sbt`:
    ```scala
    import sbtassembly.AssemblyPlugin.defaultShellScript
    assemblyOption in assembly := (assemblyOption in assembly).value.copy(prependShellScript = Some(defaultShellScript))
    ```

2. Assemble on Windows: `sbt assembly`

3. Run on Linux/OS X:
    ```
    $ ./foo-assembly-0.0.1.jar
    : No such file or directory
    ```

I understand that a simple solution would be to assemble on the target platform, but this isn't always practical.

Do you have a developer guide or any hints about the following errors I'm seeing when running `scripted` tests?

1. Repo errors:
    ```
    [info] Resolving org.eclipse.jgit#org.eclipse.jgit-parent;3.7.0.201502260915-r ...
    [warn] Host repo.lessis.me not found. url=http://repo.lessis.me/org/eclipse/jgit/org.eclipse.jgit-parent/3.7.0.201502260915-r/org.eclipse.jgit-parent-3.7.0.201502260915-r.jar
    [info] You probably access the destination server through a proxy server that is not well configured.
    [error] Server access Error: Connection timed out url=http://repo.codahale.com/org/eclipse/jgit/org.eclipse.jgit-parent/3.7.0.201502260915-r/org.eclipse.jgit-parent-3.7.0.201502260915-r.jar
    ```

2. Errors compiling scala-2.9.1.final compiler-interface (I'm using Java 1.8.0_73):
    ```
    [info] [info] 'compiler-interface' not yet compiled for Scala 2.9.1.final. Compiling...
    [error] error: error while loading CharSequence, class file '/usr/java/jdk1.8.0_73/jre/lib/rt.jar(java/lang/CharSequence.class)' is broken
    [error] (bad constant pool tag 15 at byte 1501)
    [error] error: error while loading AnnotatedElement, class file '/usr/java/jdk1.8.0_73/jre/lib/rt.jar(java/lang/reflect/AnnotatedElement.class)' is broken
    [error] (bad constant pool tag 15 at byte 2713)
    [error] error: error while loading Arrays, class file '/usr/java/jdk1.8.0_73/jre/lib/rt.jar(java/util/Arrays.class)' isbroken
    [error] (bad constant pool tag 15 at byte 12821)
    [error] error: error while loading Comparator, class file '/usr/java/jdk1.8.0_73/jre/lib/rt.jar(java/util/Comparator.class)' is broken
    [error] (bad constant pool tag 15 at byte 5003)
    [error] /tmp/sbt_92966ec6/xsbt/ExtractAPI.scala:479: error: java.util.Comparator does not take type parameters
    [error]   private[this] val sortClasses = new Comparator[Symbol] {
    [error]                                       ^
    [error] 5 errors found
    [info] [error] (compile:compileIncremental) Error compiling sbt component 'compiler-interface'
    [info] [error] Total time: 6 s, completed Apr 29, 2016 6:36:31 PM
    ```

--- a/src/main/scala/sbtassembly/Assembly.scala
+++ b/src/main/scala/sbtassembly/Assembly.scala
@@ -42,7 +42,7 @@ object Assembly {
         val tmpFile = cacheDir / "assemblyExec.tmp"
         if (tmpFile.exists()) tmpFile.delete()
         val jarCopy = IO.copyFile(outPath, tmpFile)
-        IO.writeLines(outPath, shellScript, append = false)
+        IO.write(outPath, shellScript.mkString("\n"), append = false)
         val jarBytes = IO.readBytes(tmpFile)
         tmpFile.delete()
         IO.append(outPath, jarBytes)

